internal/integrationtest/daemon/detect_core_changes_test.go

```diff
@@ -0,0 +1,125 @@
+// This file is part of arduino-cli.
+//
+// Copyright 2022 ARDUINO SA (http://www.arduino.cc/)
+//
+// This software is released under the GNU General Public License version 3,
+// which covers the main part of arduino-cli.
+// The terms of this license can be found at:
+// https://www.gnu.org/licenses/gpl-3.0.en.html
+//
+// You can be released from the requirements of the above licenses by purchasing
+// a commercial license. Buying such a license is mandatory if you want to
+// modify or otherwise use the software for commercial activities involving the
+// Arduino software without disclosing the source code of your own applications.
+// To purchase a commercial license, send an email to license@arduino.cc.
+
+package daemon_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"testing"
+	"time"
+
+	"github.com/arduino/arduino-cli/internal/integrationtest"
+	"github.com/arduino/arduino-cli/rpc/cc/arduino/cli/commands/v1"
+	"github.com/arduino/go-paths-helper"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestDetectionOfChangesInCoreBeforeCompile(t *testing.T) {
```

Comment:
Makes sense I've updated the test so it touches each file you listed and checks that the compile detects the change correctly.

About file-delete and file-create it tests only the `programmers.txt` since it's the only file that we can delete while keeping the platform working. Since the `TimestampStore` uses the same subroutines for all the files, testing file-delete and file-create only on `programmers.txt` gives high confidence that it will work on the other files too so, complicating the test more do not bring much more value IMHO.